### PR TITLE
Refactors ensembler, online-ensembler-forecaster and descendants

### PR DESCRIPTION
**Pull request by thayeylolu**
Reference : https://github.com/alan-turing-institute/sktime/issues/955

We need to slowly re-factor all forecasters to be compliant with the new interface (see PR #912). i.e., no overrides of fit etc and descendant classes only implementing core logic, while the BaseForecaster implements check etc logic.

Currently this is running due to various downwards compatibility workarounds, which should be removed once all forecasters are ready (in "no-override" state)

The forecasters that still need a downstream refactor are:

  

- [x] Online Ensembler Forecaster , Descendant and Ensembler

### CODE REVIEW

File: sktime/forecasting/online_learning/_online_ensemble.py [hunk 1]
```diff
@@ -47,15 +52,12 @@ def fit(self, y, X=None, fh=None):
...
-        self._set_y_X(y, X)
-        self._set_fh(fh)
         names, forecasters = self._check_forecasters()
         self.weights = np.ones(len(forecasters)) / len(forecasters)
         self._fit_forecasters(forecasters, y, X, fh)
-        self._is_fitted = True
         return self
 
-    def _fit_ensemble(self, y, X=None):
+    def fit_ensemble(self, y, X=None):
```
Comment (mloning) [inline sktime/forecasting/online_learning/_online_ensemble.py:60]: This should still be private, no?
```suggestion
    def _fit_ensemble(self, y, X=None):
```

**Comment by thayeylolu** [comment]
Could you point me to where you are referring to @fkiraly

**Review by mloning — COMMENTED** [review]
Hi @thayeylolu, we also need to update the `update` method by removing the following I think: 
```python
        self.check_is_fitted()
        self._update_y_X(y, X)
```

File: sktime/forecasting/online_learning/_online_ensemble.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/online_learning/_online_ensemble.py:60]: yes, why was this changed?

**Comment by fkiraly** [comment]
>Could you point me to where you are referring to @fkiraly

the `update`/`_update` function, currently line 59.

File: sktime/forecasting/online_learning/_online_ensemble.py (hunk 1 above)
Comment (thayeylolu) [reply sktime/forecasting/online_learning/_online_ensemble.py]: Must have been an error

**Review by fkiraly — APPROVED** [review]
looks ready to merge, I'll just wait for the checks to finish.

**Pull request merged**